### PR TITLE
Add support for @Unsafe to analyzer

### PR DESCRIPTION
**Pull request by mehmetf**
Add support for `@Unsafe` annotation as `isUnsafe` and `hasUnsafe`.

See: https://github.com/dart-lang/sdk/pull/40429 for the definition of the annotation.

### CODE REVIEW

File: pkg/analyzer/lib/src/dart/element/element.dart [hunk 1]
```diff
@@ -2392,6 +2392,9 @@ class ElementAnnotationImpl implements ElementAnnotation {
   /// The name of the top-level variable used to mark a class as being sealed.
   static const String _SEALED_VARIABLE_NAME = "sealed";
 
+  /// The name of the top-level variable used to mark a method as unsafe.
```
Comment (pq) [inline pkg/analyzer/lib/src/dart/element/element.dart]: Is this just methods or other members?

File: pkg/analyzer/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+## 0.39.6
```
Comment (pq) [inline pkg/analyzer/CHANGELOG.md]: I *think* this should be 0.39.5?

/cc @stereotype441

File: pkg/analyzer/lib/src/dart/element/element.dart (hunk 1 above)
Comment (bwilkerson) [reply pkg/analyzer/lib/src/dart/element/element.dart]: nit: `Unsafe` is a class, not a variable.

File: pkg/analyzer/lib/src/dart/element/element.dart [hunk 3]
```diff
@@ -2540,6 +2543,12 @@ class ElementAnnotationImpl implements ElementAnnotation {
       element.name == _SEALED_VARIABLE_NAME &&
       element.library?.name == _META_LIB_NAME;
 
+  @override
+  bool get isUnsafe =>
+      element is PropertyAccessorElement &&
```
Comment (bwilkerson) [inline pkg/analyzer/lib/src/dart/element/element.dart]: This seems wrong to me. I would have expected that it would be a `ClassElement`.

**Review by bwilkerson — COMMENTED** [review]
We obviously need to resolve the question of where the annotation lives before committing this.

File: pkg/analyzer/lib/src/dart/element/element.dart (hunk 3 above)
Comment (mehmetf) [reply pkg/analyzer/lib/src/dart/element/element.dart]: Hint: I don't know what I am doing :-) So thanks for all the help!

I have not setup my environment to test this yet. When the annotation is agreed upon and pushed to master, I can tinker with this.

File: pkg/analyzer/CHANGELOG.md (hunk 2 above)
Comment (mehmetf) [reply pkg/analyzer/CHANGELOG.md]: Yeah, I am not sure what `-dev` means....

File: pkg/analyzer/lib/src/dart/element/element.dart (hunk 1 above)
Comment (mehmetf) [reply pkg/analyzer/lib/src/dart/element/element.dart]: Fixed.

**Comment by mehmetf** [comment]
Closing this PR as there's a better proposal being discussed.

**Pull request closed without merging**